internal/replication/blockreplicator.go

```diff
@@ -246,6 +267,67 @@ func (br *BlockReplicator) Start() {
 	<-readyProposeCh
 }
 
+// startOnBoarding pulls the missing blocks from the current ledger height up to (and including) the join block.
+// It then constructs the raftpb.ConfState and raftpb.Snapshot from the cluster configuration of the join block and
+// stores it. This way, raft storage starts with the right state to join the cluster. In addition, if the node is
+// restarted, the snapshot indicates that the node needs to restart normally, rather than perform on-boarding again.
+func (br *BlockReplicator) startOnBoarding() {
+	br.lg.Info("Starting the on-boarding process")
+
+	err := br.onBoard(br.joinBlock)
+	if err != nil {
+		switch err.(type) {
+		case *ierrors.ClosedError:
+			br.lg.Warn("Closing, stopping on-boarding")
+			return // do not cause panic when the server shuts down
+		default:
+			br.lg.Panicf("Failed to on-board to join-block number: %d, block: %+v", br.joinBlockNumber, br.joinBlock)
+		}
+	}
+
+	br.lg.Infof("On-boarding completed successfully, starting replication")
+	genesisBlock, err := br.ledgerReader.Get(1)
+	if err != nil {
+		br.lg.Panicf("Failed to read genesis block after on-boarding")
+	}
+	genesisConfig := genesisBlock.GetConfigTxEnvelope().GetPayload().GetNewConfig()
+	br.lg.Debugf("Genesis config: %v", genesisConfig)
```

Comment:
Why do we need access genesis block config here, I don't see it used anywhere. If it for debug purpose, lets enclosure it in some `if`

Reply:
That is correct. It is part of debug code I forgot to remove.